sql/core/src/test/scala/org/apache/spark/sql/test/TestSQLContext.scala

```diff
@@ -47,6 +51,10 @@ private[spark] class TestSparkSession(sc: SparkContext) extends SparkSession(sc)
     testData.loadTestData()
   }
 
+  override def newSession(): TestSparkSession = {
+    new TestSparkSession(sparkContext, Some(sharedState))
+  }
+
   private object testData extends SQLTestData {
```

Comment:
Can we reuse `testData` even creating a fresh copy of the spark session? Seems tests can reuse `testData`?

Reply:
There are only 3 java test suites that rely on it, and I don't think we would apply `SharedSparkSessionWithFreshCopy` for them. They don't even extend `SharedSparkSession`.